linalg/solvers.hpp

```diff
@@ -473,6 +473,38 @@ class SLBQPOptimizer : public OptimizationSolver
    virtual void Mult(const Vector &xt, Vector &x) const;
 };
 
+class BlockILU0 : public Solver
+{
+public:
+   BlockILU0(int block_size_ = 1, bool reorder_ = true);
```

Comment:
I'm not sure `block_size` of 1 is a sensible default here. I think it makes more sense to have no default and require this parameter (it is a *block* ILU preconditioner).

Reply:
Agreed.